src/main/resources/swagger.api/folio-modules/srs/record/sourceRecord.json

```diff
@@ -0,0 +1,92 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Record DTO Schema",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "id": {
+      "description": "UUID",
+      "$ref": "../../common/uuid.json"
+    },
+    "snapshotId": {
+      "description": "Corresponding snapshot id, which is the same as jobExecutionId",
+      "$ref": "../../common/uuid.json"
+    },
+    "matchedId": {
+      "description": "Represents relations with another matched records, necessary for saving the history of applying changes for inventory/holding/etc records",
+      "$ref": "../../common/uuid.json"
+    },
+    "generation": {
+      "description": "Generation from the last record with the same matchedId incremented by 1. Starts from 0.",
+      "type": "integer"
+    },
+    "recordType": {
+      "description": "Type of record, e.g. MARC_BIB",
+      "type": "string",
+      "$ref": "sourceRecordType.json"
+    },
+    "rawRecord": {
```

Comment:
In this file, we only need to define the properties we're interested in, correct? Since FAIL_ON_UNKNOWN_PROPERTIES is set to false in the ObjectMapper, Jackson will automatically ignore any properties in the event that don't have a corresponding field in the Java class.

A smaller class has a couple of advantages:

1. We avoid the need to maintain these unnecessary fields & files. (For example, we do not need to maintain rowRecord.json file)
2. During deserialization, the application won’t need to map unused properties into java fields & objects, which are discarded anyway. This can improve both performance and memory usage.

What do you think?

Reply:
I thiught it could be useful to have all the fields.. but I see you're right, because of YAGNI principle also.
Removed all unnecessary fields, but some could be useful in near future:
- deleted flag
- additionalInfo with suppressDiscovery flag
- state (probably we should filter by the this field during update ops)